swap/peer.go

```diff
@@ -151,19 +179,22 @@ func (p *Peer) createCheque() (*Cheque, error) {
 }
 
 // sendCheque sends a cheque to peer
-// To be called with mutex already held
-// Caller must be careful that the same resources aren't concurrently read and written by multiple routines
+// if there is already a pending cheque it will resend that one
+// otherwise it will create a new cheque and save it as the pending cheque
+// the caller is expected to hold p.lock
 func (p *Peer) sendCheque() error {
+	if p.getPendingCheque() != nil {
+		p.logger.Warn("old cheque still pending, resending cheque")
```

Comment:
done

Reply:
i don't think we should add a `*Cheque` variable to the log context.

when printing a variable of this type, the output looks like this:
>`&{{[240 93 250 188 76 139 45 65 27 86 59 246 18 90 204 174 252 90 81 106] [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 19633281260515} 19633281260515 [170 141 218 14 230 167 102 20 221 53 114 74 74 123 113 37 255 31 40 14 18 27 171 43 213 27 236 60 233 172 37 169 80 195 229 153 42 157 97 145 134 16 183 144 84 25 249 227 152 68 104 218 22 62 84 106 127 11 54 79 48 228 91 204 27]}`

i doubt it will be human-readable in the logs.

maybe we could take a page out of `peer.go:206`:
>`p.logger.Info("sending cheque to peer", "honey", cheque.Honey, "cumulativePayout", cheque.ChequeParams.CumulativePayout, "beneficiary", cheque.Beneficiary, "contract", cheque.Contract)`

Reply:
Implemented `String()` on `Cheque` instead.

Reply:
even better!